account_payment_term_extension/models/account_payment_term.py

```diff
@@ -0,0 +1,216 @@
+                raise exceptions.Warning(_("Payment days field format is not valid."))
+
+    payment_days = fields.Char(
+        string="Payment day(s)",
+        help="Put here the day or days when the partner makes the payment. "
+        "Separate each possible payment day with dashes (-), commas (,) "
+        "or spaces ( ).",
+    )
+
+
+class AccountPaymentTerm(models.Model):
+    _inherit = "account.payment.term"
+
+    sequential_lines = fields.Boolean(
+        string="Sequential lines",
+        default=False,
+        help="Allows to apply a chronological order on lines.",
+    )
+    holiday_ids = fields.One2many(
+        string="Holidays",
+        comodel_name="account.payment.term.holiday",
+        inverse_name="payment_id",
+    )
+
+    def apply_holidays(self, date):
+        holiday = self.holiday_ids.search(
+            [("payment_id", "=", self.id), ("holiday", "=", date)]
+        )
+        if holiday:
+            return fields.Date.from_string(holiday.date_postponed)
+        return date
+
+    def apply_payment_days(self, line, date):
+        """Calculate the new date with days of payments"""
+        if line.payment_days:
+            payment_days = line._decode_payment_days(line.payment_days)
+            if payment_days:
+                new_date = None
+                payment_days.sort()
+                days_in_month = calendar.monthrange(date.year, date.month)[1]
+                for day in payment_days:
+                    if date.day <= day:
+                        if day > days_in_month:
+                            day = days_in_month
+                        new_date = date + relativedelta(day=day)
+                        break
+                if not new_date:
+                    day = payment_days[0]
+                    if day > days_in_month:
+                        day = days_in_month
+                    new_date = date + relativedelta(day=day, months=1)
+                return new_date
+        return date
+
+    def compute(self, value, date_ref=False, currency=None):
+        """Complete overwrite of compute method to add rounding on line
+        computing and also to handle weeks and months
+        """
+        self.ensure_one()
+        date_ref = date_ref or fields.Date.today()
+        amount = value
+        result = []
+        if not currency and self.env.context.get("currency_id"):
+            currency = self.env["res.currency"].browse(self.env.context["currency_id"])
+        else:
+            currency = self.env.user.company_id.currency_id
+        prec = currency.decimal_places
+        next_date = fields.Date.from_string(date_ref)
+        for line in self.line_ids:
+            amt = line.compute_line_amount(value, amount, prec)
+            if not self.sequential_lines:
+                # For all lines, the beginning date is `date_ref`
+                next_date = fields.Date.from_string(date_ref)
+                if float_is_zero(amt, precision_rounding=prec):
```

Comment:
Either we use float_is_zero(amt, precision_rounding=currency.rounding) or we use float_is_zero(amt, precision_digits=currency.decimal_places)... but not precision_rounding with currency.decimal_places !!!
Yep, it seems I found a bug that exists since v9 ! That's because I made a test in EUR currency with an invoice with total = 0.20 euros !
Watch out when fixing this, prec is used below in a round()...

Reply:
I made a PR to fix this, that include a test on that particular bug: https://github.com/Domatix/account-payment/pull/1

Reply:
Thanks. Merged.